
From: Stephen Tweedie <sct@redhat.com>

Return EPERM, not EACCES, if we try to extend the filesystem without
sufficient privilege.

Signed-off-by: Stephen Tweedie <sct@redhat.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/fs/ext3/ioctl.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff -puN fs/ext3/ioctl.c~ext3-online-resize-fix-error-codes fs/ext3/ioctl.c
--- 25/fs/ext3/ioctl.c~ext3-online-resize-fix-error-codes	Thu Sep 30 17:32:25 2004
+++ 25-akpm/fs/ext3/ioctl.c	Thu Sep 30 17:32:25 2004
@@ -181,7 +181,7 @@ flags_err:
 		int err;
 
 		if (!capable(CAP_SYS_RESOURCE))
-			return -EACCES;
+			return -EPERM;
 
 		if (sb->s_flags & MS_RDONLY)
 			return -EROFS;
@@ -202,7 +202,7 @@ flags_err:
 		int err;
 
 		if (!capable(CAP_SYS_RESOURCE))
-			return -EACCES;
+			return -EPERM;
 
 		if (inode->i_sb->s_flags & MS_RDONLY)
 			return -EROFS;
_
